test/common/upstream/health_checker_impl_test.cc

```diff
@@ -129,6 +129,17 @@ class HttpHealthCheckerImplTest : public testing::Test {
   HttpHealthCheckerImplTest()
       : cluster_(new NiceMock<MockClusterMockPrioritySet>()),
         event_logger_(new MockHealthCheckEventLogger()) {}
+  void allocHealthChecker(envoy::config::core::v3::HealthCheck conf) {
```

Comment:
You should take `conf` by const ref to avoid an extra copy. Better yet, just take the yaml as an arg and do the parseHealthCheckFromV2Yaml inside the function.